Content.Server/Chemistry/Components/ReagentDispenserComponent.cs

```diff
@@ -7,20 +9,56 @@
 namespace Content.Server.Chemistry.Components
 {
     /// <summary>
-    /// A machine that dispenses reagents into a solution container.
+    /// A machine that dispenses reagents into a solution container from containers in its storage slots.
     /// </summary>
     [RegisterComponent]
     [Access(typeof(ReagentDispenserSystem))]
     public sealed partial class ReagentDispenserComponent : Component
     {
-
+        /// <summary>
+        /// String with the pack name that stores the initial fill of the dispenser. The initial
+        /// fill is added to the dispenser on MapInit. Note that we don't use ContainerFill because
+        /// we have to generate the storage slots at MapInit first, then fill them.
+        /// </summary>
         [DataField("pack", customTypeSerializer:typeof(PrototypeIdSerializer<ReagentDispenserInventoryPrototype>))]
         [ViewVariables(VVAccess.ReadWrite)]
         public string? PackPrototypeId = default!;
 
-        [DataField("emagPack", customTypeSerializer:typeof(PrototypeIdSerializer<ReagentDispenserInventoryPrototype>))]
-        [ViewVariables(VVAccess.ReadWrite)]
-        public string? EmagPackPrototypeId = default!;
+        /// <summary>
+        /// Maximum number of internal storage slots. Dispenser can't store (or dispense) more than
+        /// this many chemicals (without unloading and reloading).
+        /// </summary>
+        [DataField("numStorageSlots")]
```

Comment:
for most fields here, remove the string from the datafield since it just inherits the variable name in camel case.